crates/serde/src/lib.rs

```diff
@@ -48,3 +49,17 @@ where
 {
     s.serialize_str(&format!("{x:x}"))
 }
+
+/// Custom deserializer for `state_root` that treats `"0x"` or empty as `B256::ZERO`.
+pub fn deserialize_state_root<'de, D>(deserializer: D) -> Result<B256, D::Error>
```

Comment:
we can move this to the header file and keep as private fn, 
let's also reference the issue in the comment